cumulusci/tasks/bulkdata/tests/test_snowfakery.py

```diff
@@ -20,11 +21,52 @@
 
 original_refresh_token = OrgConfig.refresh_oauth_token
 
+FAKE_LOAD_RESULTS = (
+    {
+        "Insert Account": {
+            "mapping": {"sf_object": "Account"},
+            "records_processed": 2,
+            "total_row_errors": 0,
+        },
+        "Insert Contact": {
+            "mapping": {"sf_object": "Contact"},
+            "records_processed": 2,
+            "total_row_errors": 0,
+        },
+    },
+    {
+        "Insert Account": {
+            "mapping": {"sf_object": "Account"},
+            "records_processed": 3,
+            "total_row_errors": 0,
+        },
+        "Insert Contact": {
+            "mapping": {"sf_object": "Contact"},
+            "records_processed": 3,
+            "total_row_errors": 0,
+        },
+    },
+)
+
+
+def call_closure():
+    return_values = cycle(iter(FAKE_LOAD_RESULTS))
+
+    def __call__(self, *args, **kwargs):
+        self.return_values = {"step_results": next(return_values)}
+        return __call__.mock(*args, **kwargs)
+
+    __call__.mock = mock.Mock()
+
+    return __call__
```

Comment:
This is a lot of magic for test code. Please add a comment about how it's here to cycle through the fake results each time the task is called.

I don't understand the part about setting `__call__.mock` and using it within the closure. `__call__` doesn't actually need to return anything, does it?

Reply:
I want lines like this to work:

        assert len(mock_load_data.mock_calls) == 3
        
 The gist is that I'm sort of wrapping a mock so I can inject a side_effect. The normal side effect techniques are hard to achieve with pure mocks because they don't get bound like bound methods do. So I need a "real function" that wraps the mock.
 
 I can add docs to that effect.

Reply:
It might be clearer as a purpose-built fake object that updates return_values and records the call count rather than trying to wrap the mock. But I understand now and I'll accept it if you add a comment.